From 135a9f4b5aead507c030fb7e3e8ad13aaa91f403 Mon Sep 17 00:00:00 2001
From: Daiki Ueno <ueno@gnu.org>
Date: Fri, 07 Nov 2014 02:31:12 +0000
Subject: epg: Adjust to GnuPG 2.1 key listing change

* epg.el (epg--list-keys-1): Ignore fields after the 15th field
(bug#18979).  Reported by Hideki Saito.
---
(limited to 'lisp/epg.el')

diff --git a/lisp/epg.el b/lisp/epg.el
index 0c74531..e4d8c1e 100644
--- a/lisp/epg.el
+++ b/lisp/epg.el
@@ -1278,8 +1278,9 @@ callback data (if any)."
 	      string (match-string 0)
 	      index 0
 	      field 0)
-	(while (eq index
-		   (string-match "\\([^:]+\\)?:" string index))
+	(while (and (< field (length (car keys)))
+		    (eq index
+			(string-match "\\([^:]+\\)?:" string index)))
 	  (setq index (match-end 0))
 	  (aset (car keys) field (match-string 1 string))
 	  (setq field (1+ field))))
--
cgit v0.9.0.2
